### PR TITLE
Handle Nested REST APIs

### PR DESCRIPTION
**Pull request by jimdillon**
* Pass Parameters through to adapter functions to allow for nested REST APIs

@evanshortiss Thoughts?

### CODE REVIEW

**Comment by evanshortiss** [comment]
I'm currently working on an update to the module that this PR will align very nicely with! Let me push the changes ASAP and I'll ping you so you can rebase and we can get this merged 👍

File: lib/router.js [hunk 1]
```diff
@@ -124,6 +124,7 @@ module.exports = function (dataset, adapter) {
       adapter,
       'list',
       {
+        params: req.params,
```
Comment (evanshortiss) [inline lib/router.js]: I'm thinking we might want to call this _routeParams_, just to be explicit about their nature.

**Comment by evanshortiss** [comment]
@jimdillon can you update this with the latest code base and perhaps rename the _params_ (see PR comment). I see no conflict warnings but might be required now that 0.2.0 is out. Let me know when you're done.

**Comment by jimdillon** [comment]
Will do right after lunch.

**Comment by jimdillon** [comment]
@evanshortiss, variable changed to routeParams as requested.

**Pull request merged**